lib/item.rb

```diff
@@ -0,0 +1,32 @@
+require 'date'
+
+class Item
+  attr_reader :id
+  attr_accessor :publish_date, :archived
+
+  def initialize(publish_date, archived: false)
+    @id = Random.rand(1..1000)
+    @publish_date = Date.parse(publish_date)
+    @archived = archived
+  end
+
+  def move_to_archive
+    @archived = can_be_archived?
+  end
+
+  def label=(label)
+    @label = label
+    label.items << self
```

Comment:
- [ ] @Fatima-hub333 great work so far, in this function you need to ensure the Item class has only one label or author, so your method should check if the item has the label before adding a label into it.(you can use unless to check if the label is already there, if not then it adds it).

Reply:
Done